linux/keyman-system-service/build.sh

```diff
@@ -35,6 +35,39 @@ else
 fi
 MESON_PATH="build/$(uname -m)/$MESON_TARGET"
 
+clean_action() {
+  rm -rf "$THIS_SCRIPT_PATH/build/"
+}
+
+check_missing_coverage_configuration() {
+  if builder_has_option --coverage && [ -d "$MESON_PATH" ]; then
+    # It's possible that we got configured because we're a dependency of ibus-keyman
+    # in which case the `--coverage` option wasn't passed along.
+    cd "$MESON_PATH"
+    if ! ninja -t targets | grep -q coverage-html ; then
+      cd "$THIS_SCRIPT_PATH"
+      clean_action
+    fi
+    cd "$THIS_SCRIPT_PATH"
+  fi
```

Comment:
Is there an easier solution? I don't think our `build.sh` system allows to pass the `--coverage` option to dependencies, or does it?

Reply:
keyman-system-service should be a child build rather than a dependency. A dependency is explicitly something that is required for the project but that you only want to build and use. A child build is a component owned by the parent build.

https://github.com/keymanapp/keyman/blob/1e734a47f5337515b58cfdc55145bc5dd4de2660/resources/build/build-utils.md#L447-L451

Reply:
It's both. You can't run ibus-keyman tests without first building keyman-system-service.

Or do you mean keyman-system-service should be a child of ibus-keyman? But that doesn't make completely sense either because it's not really a component of ibus-keyman.

Reply:
I'm not really sure what you are trying to solve here. It seems like the top-level linux/build.sh is already building all the modules as child builds, and the `--coverage` option already has `+` so it should be passed through.

If things are building in the wrong order, you could try putting the keyman-system-service first in linux/build.sh.